### ISSUE TITLE
Ripgrep pre-filtering step fails if the match template doesn't find anything

### ISSUE THREAD

**Issue by kkom** [comby-tools/comby#283]
**Describe the bug**

It seems that the -rg pre-filtering step fails if the comby match template doesn't find anything.

**Reproducing**

Take a look at this example repo:

```
➜  my-repo git:(master) cat gradle/wrapper/gradle-wrapper.properties 
#Thu Jan 16 14:46:50 CET 2020
distributionUrl=https\://services.gradle.org/distributions/gradle-5.2.1-all.zip
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStorePath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
```

Note that `gradle-5.2.1` will not match the match template. Now see how comby behaves if you add the ripgrep pre-filtering step:

```
➜  my-repo git:(master) comby \
        -matcher .generic \
        -rg "" \       
        "distributionUrl=:[urlPrefix]gradle-6.:[~[\.\d]*]:[urlSuffix]" \
        "distributionUrl=:[urlPrefix]gradle-6.8.3:[urlSuffix]"
Error executing rg, exit status 1.
```

**Expected behavior**

This is an example of a repository where the comby match template will match something:

```
➜  my-repo git:(master) ✗ cat gradle/wrapper/gradle-wrapper.properties
#Tue Jan 21 10:15:56 CET 2020
distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.3-all.zip
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStorePath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
➜  my-repo git:(master) ✗ comby \                                     
        -matcher .generic \
        -rg "" \
        "distributionUrl=:[urlPrefix]gradle-6.:[~[\.\d]*]:[urlSuffix]" \
        "distributionUrl=:[urlPrefix]gradle-6.8.3:[urlSuffix]"
------ gradle/wrapper/gradle-wrapper.properties
++++++ gradle/wrapper/gradle-wrapper.properties
@|-1,5 +1,5 ============================================================
 |#Tue Jan 21 10:15:56 CET 2020
-|distributionUrl=https\://services.gradle.org/distributions/gradle-6.7.3-all.zip
+|distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
 |distributionBase=GRADLE_USER_HOME
 |distributionPath=wrapper/dists
 |zipStorePath=wrapper/dists
```

Everything is fine even if the change is a no-op:

```
➜  my-repo git:(master) cat gradle/wrapper/gradle-wrapper.properties             
#Tue Jan 21 10:15:56 CET 2020
distributionUrl=https\://services.gradle.org/distributions/gradle-6.8.3-all.zip
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStorePath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
➜  my-repo git:(master) comby \                                                  
        -matcher .generic \
        -rg "" \
        "distributionUrl=:[urlPrefix]gradle-6.:[~[\.\d]*]:[urlSuffix]" \
        "distributionUrl=:[urlPrefix]gradle-6.8.3:[urlSuffix]"
➜  my-repo git:(master) 
```

**Additional context**

```
➜  ~ comby --version
1.5.0
➜  ~ rg --version
ripgrep 12.1.1
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)
```

**Comment by rvantonder** [comment]
Fixed this in 1.5.1 to not emit an error. Note that `comby` will exit with exit code 0 even for no matches with and without `-rg` (grep and rg will traditionally exit with code 1 for no matches). Please reopen if there's a different expectation.

**Comment by kkom** [comment]
Thanks for fixing it! :) No concerns around that 0 return code, at least for the use cases I came across so far
